ilp-core/src/test/java/org/interledger/core/InterledgerAddressBuilderTest.java

```diff
@@ -72,46 +69,22 @@ public void testConstruction_LedgerPrefix() {
 
   @Test(expected = IllegalArgumentException.class)
   public void test_empty_address() {
-    final String value = "";
-    try {
-      InterledgerAddress.builder().value(value).build();
-    } catch (IllegalArgumentException e) {
-      assertThat(e.getMessage(), is(String.format(EXPECTED_ERROR_MESSAGE, value)));
-      throw e;
-    }
+    InterledgerAddress.builder().value("").build();
   }
 
   @Test(expected = IllegalArgumentException.class)
   public void test_blank_address() {
-    final String value = "  ";
-    try {
-      InterledgerAddress.builder().value(value).build();
-    } catch (IllegalArgumentException e) {
-      assertThat(e.getMessage(), is(String.format(EXPECTED_ERROR_MESSAGE, value)));
```

Comment:
I think we should keep the error message assertions. Primarily, it validates that the `IllegalArgumentException` that was thrown was actually what we expected, as opposed to being some _other_ IllegalArgumentException. 

Thoughts?

Reply:
I don't think so. Here is my point of view, through another domain use-case:

`doorKeeper#canStartBlackFriday` // possible catches: `goodsUnavailableException(pendingDeliveryException)`, `goodsUnavailableException(supplierOutOfStockException)`, `tooEarlyException("Disallowed before 9AM")`
__-> `warehouseManager#areGoodsAvailable` // possible catches: `pendingDeliveryException, supplierOutOfStockException`
____-> `transportPartner#isOrderDelivered`
____-> `supplier#isSerialNumberInStock`
__-> `timeService#isGTE9AM`

By separating concerns, the `doorKeeper` may:
* Either, delay the doors opening until all lights are green
* Either, contact the `storeOwner` and forward him the underlying `goodsUnavailableException` or `tooEarlyException` for decision
* Either, delay the doors opening in case of a `goodsUnavailableException`
* Either, bypass restriction with a tolerance threshold in case of a `tooEarlyException`

That is, he has a shallow view of the situation, and should probably not be given a deeper visibility on nested events (i.e. knowing that goods are not available due to a supplier shortage or to a transport issue is too much knowledge for his business imo).

Put back in our situation, PR is currently:

`test-caller` // catches: `illegalArgumentException("Address has no scheme prefix")`
__->addressBuilder#build // passes through the unaltered: `illegalArgumentException("Address has no scheme prefix")`
____-> `addressParser#validate`

Just now, I would change this into:

`test-caller` // catches: `illegalArgumentException("Address is invalid", illegalArgumentException)` (and we may imagine other causes such as `"Address has no route to destination"` even if it seems unsuitable to be managed so early at runtime)
__-> `addressBuilder#build` // passes through: `illegalArgumentException("Address has no scheme prefix")`
____-> \_ `addressParser#validate`

And the `addressBuilder` test-class would check for the `Address is invalid` error message, rather than digging into the parser internals (which are already checked in the parser test-class, therefore would be redundant in here).

In the long run, rather than relying on error messages only, I would subclass `InterledgerRuntimeException` into `InvalidAddressException`, `AnotherCauseException` and so on.
`test-caller` would then shallow-check the exception type only.

Reply:
Anyway, I can do as you suggested (which also ends up in redundancy of the address-invalidity cause check among test-classes), just confirm and I'll do so.

Reply:
This point is now adressed: I restored the `IllegalArgumentException` checks in test-classes that use the `InterledgerAddressBuilder`.